server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -1211,88 +1005,66 @@ public void testSimpleParserNoTypeNoId() throws Exception {
         assertThat((String) builtDocMapper.meta().get("param1"), equalTo("value1"));
     }
 
     public void testNoDocumentSent() throws Exception {
-        IndexService indexService = createIndex("test");
-        DocumentMapper docMapper = new DocumentMapper.Builder(
-                new RootObjectMapper.Builder("person")
-                        .add(new ObjectMapper.Builder("name")
-                            .add(new TextFieldMapper.Builder("first").store(true).index(false))),
-            indexService.mapperService()).build(indexService.mapperService());
-
+        DocumentMapper docMapper = createDocumentMapper(mapping(b -> {}));
         BytesReference json = new BytesArray("".getBytes(StandardCharsets.UTF_8));
-        try {
-            docMapper.parse(new SourceToParse("test", "1", json, XContentType.JSON)).rootDoc();
-            fail("this point is never reached");
-        } catch (MapperParsingException e) {
-            assertThat(e.getMessage(), equalTo("failed to parse, document is empty"));
-        }
+        MapperParsingException e = expectThrows(MapperParsingException.class,
+            () -> docMapper.parse(new SourceToParse("test", "1", json, XContentType.JSON)));
+        assertThat(e.getMessage(), equalTo("failed to parse, document is empty"));
     }
 
     public void testNoLevel() throws Exception {
-        String defaultMapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type").endObject().endObject());
-
-        DocumentMapper defaultMapper = createIndex("test").mapperService().documentMapperParser()
-            .parse("type", new CompressedXContent(defaultMapping));
-
-        ParsedDocument doc = defaultMapper.parse(new SourceToParse("test", "1", BytesReference
-                .bytes(XContentFactory.jsonBuilder()
-                        .startObject()
-                        .field("test1", "value1")
-                        .field("test2", "value2")
-                        .startObject("inner").field("inner_field", "inner_value").endObject()
-                        .endObject()),
-                XContentType.JSON));
-
+        DocumentMapper defaultMapper = createDocumentMapper(mapping(b -> {}));
+        ParsedDocument doc = defaultMapper.parse(source(b -> {
+            b.field("test1", "value1");
+            b.field("test2", "value2");
+            b.startObject("inner").field("inner_field", "inner_value").endObject();
+        }));
         assertThat(doc.rootDoc().get("test1"), equalTo("value1"));
         assertThat(doc.rootDoc().get("test2"), equalTo("value2"));
         assertThat(doc.rootDoc().get("inner.inner_field"), equalTo("inner_value"));
     }
 
-    public void testTypeLevel() throws Exception {
-        String defaultMapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type").endObject().endObject());
+    // TODO do we still need all this tests for 'type' at the bottom level now that
+    // we no longer have types?
```

Comment:
There are a bunch of tests here that check that if you have a field called `foo` it doesn't get confused if you also have a type called `foo`.  Seeing as we don't have types anymore, apart from `_doc`, and I don't think that you can actually create a field called `_doc` I'm not sure they're strictly necessary any more, but I've kept them in for now.

Reply:
I'm not sure! we do still support indices from 7.x which have types, right?

Reply:
We do - I'll keep them in for now then.